### ISSUE TITLE
Support `binding_array` of Storage Buffers on Metal

### ISSUE THREAD

**Issue by cwfitzgerald** [gfx-rs/wgpu#6741]
Once #3334 is in, we should be able to easily extend this to Storage Buffers.

**Comment by LeonardoTemperanza** [comment]
I was wondering if there's been any recent progress on this?

**Comment by cwfitzgerald** [comment]
No, there hasn't been - haven't had a ton of time to work on open source of late.
